src/sentry/deletions/defaults/organization.py

```diff
@@ -68,4 +70,7 @@ def mark_deletion_in_progress(self, instance_list):
 
         for instance in instance_list:
             if instance.status != OrganizationStatus.DELETION_IN_PROGRESS:
-                instance.update(status=OrganizationStatus.DELETION_IN_PROGRESS)
+                # Open a transaction to ensure that the outbox and model save are in sync
+                with transaction.atomic():
```

Comment:
I didn't see any wrapping transaction but I'm not 100% sure. I also figured we wanted this around each instance save so that the celery task could restart and continue if an org update or outbox insert fails